src/pages/apps/[systemId]/[appId]/launch.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * @author psarando
+ *
+ * A page for displaying the App Launch Wizard for an app with the given IDs.
+ * Also handles calling analysis submission and quick-launch-save endpoints.
+ */
+import React from "react";
+
+import { useRouter } from "next/router";
+import { useQuery } from "react-query";
+
+import constants from "../../../../constants";
+
+import NavigationConstants from "../../../../common/NavigationConstants";
+
+import { useConfig } from "../../../../contexts/config";
+import { useUserProfile } from "../../../../contexts/userProfile";
+
+import { submitAnalysis } from "../../../../serviceFacades/analyses";
+import { getAppDescription } from "../../../../serviceFacades/apps";
+
+import AppLaunchWizard from "../../../../components/apps/launch/AppLaunchWizard";
+
+export default () => {
+    const [appKey, setAppKey] = React.useState(null);
+    const [app, setApp] = React.useState(null);
+    const [appError, setAppError] = React.useState(null);
+
+    const [config] = useConfig();
+    const [userProfile] = useUserProfile();
+
+    const router = useRouter();
+    const { systemId, appId } = router.query;
+
+    React.useEffect(() => {
+        if (systemId && appId) {
+            setAppKey([
+                "getAppDescription",
+                {
+                    systemId,
+                    appId,
+                },
+            ]);
+        }
+    }, [systemId, appId]);
+
+    const { status: detailsStatus } = useQuery({
+        queryKey: appKey,
+        queryFn: getAppDescription,
+        config: {
+            onSuccess: setApp,
+            onError: setAppError,
+        },
+    });
+
+    const loading = detailsStatus === constants.LOADING;
+
+    // FIXME: notify, defaultOutputDir, and startingPath
+    // need to come from user prefs.
+    const notify = false;
+    const irodsHomePath = config?.irods?.home_path;
+    const username = userProfile?.id;
+
+    const homePath =
+        irodsHomePath && username ? `${irodsHomePath}/${username}` : "";
+    const startingPath = homePath || "";
+    const defaultOutputDir = homePath ? `${homePath}/analyses` : "";
+
+    return (
+        <AppLaunchWizard
+            notify={notify}
+            defaultOutputDir={defaultOutputDir}
+            startingPath={startingPath}
+            app={app}
+            appError={appError}
+            loading={loading}
+            submitAnalysis={(submission, onSuccess, onError) => {
+                setAppError(null);
+
+                submitAnalysis(submission)
+                    .then((resp) => {
+                        onSuccess(resp);
+                        router.push(`/${NavigationConstants.ANALYSES}`);
+                    })
+                    .catch((error) => {
+                        onError(error);
+                        setAppError(error);
+                    });
+            }}
```

Comment:
I haven't reviewed this PR. But I think we should use `react-query` here. Also, instead of navigating to `Analyses` listing here, we should navigate to job submission confirmation page. i.e `/apps/<id>/launch/submitted`. This page will show how many analyses are currently in submitted / running state along with their quota.
This page will provide users with option to 
- Go back to app listing 
- Launch another one of the same app 
- Goto analyses listing 
- Save this as quick launch. 
In future, this page can show live status of this analysis, load status of condor, k8, HPC OSG resources.

Reply:
Ohh all this is not needed for this PR. For now navigating to Analyses using `react-query` is good.